### PR TITLE
chore: upgrade to Rust 1.80.0

### PR DESCRIPTION
**Pull request by ematipico**

### CODE REVIEW

**Review by anonrig — APPROVED** [review]

File: crates/biome_cli/src/service/mod.rs [hunk 1]
```diff
@@ -72,11 +72,11 @@ type JsonRpcResult = Result<Box<RawValue>, TransportError>;
 /// This structs holds an instance of the `tokio` runtime, as well as the
 /// following fields:
 /// - `write_send` is a sender handle to the "write channel", an MPSC channel
-/// that's used to queue up requests to be sent to the server (for simplicity
-/// the requests are pushed to the channel as serialized byte buffers)
+///     that's used to queue up requests to be sent to the server (for simplicity
```
Comment (Conaclos) [inline crates/biome_cli/src/service/mod.rs:75]: Just wondering: does the rule require 4 spaces? Or 2 spaces could be enough?

File: crates/biome_js_syntax/src/generated/nodes.rs [hunk 2]
```diff
@@ -15,9 +15,7 @@ use biome_rowan::{
     AstNodeList, AstNodeListIterator, AstNodeSlotMap, AstSeparatedList,
     AstSeparatedListNodesIterator,
 };
-#[cfg(feature = "serde")]
```
Comment (Conaclos) [inline crates/biome_js_syntax/src/generated/nodes.rs:18]: Why this attribute is no longer requuired?

**Review by Conaclos — APPROVED** [review]

File: crates/biome_js_syntax/src/generated/nodes.rs (hunk 2 above)
Comment (ematipico) [reply crates/biome_js_syntax/src/generated/nodes.rs:18]: 1.80.0 got better at catching the features. This one was incorrectly set, and it's not needed. In fact, all the other syntax crates didn't have one.

File: crates/biome_cli/src/service/mod.rs (hunk 1 above)
Comment (ematipico) [reply crates/biome_cli/src/service/mod.rs:75]: Two spaces. I used the tab to be fast

**Pull request merged**